File: client-support/src/main/java/io/github/ruedigerk/contractfirst/generator/client/internal/MediaTypes.java
package io.github.ruedigerk.contractfirst.generator.client.internal;

import okhttp3.MediaType;

/**
 * Helper methods for working with media types.
 */
public class MediaTypes {

  private MediaTypes() {
    // Only static methods
  }

  public static MediaType parseNullable(String mediaType) {
    return mediaType == null ? null : MediaType.parse(mediaType);
  }

  public static boolean isJsonMediaType(String mediaType) {
    return isJsonMediaType(parseNullable(mediaType));
  }

  public static boolean isJsonMediaType(MediaType mediaType) {
    if (mediaType == null) {
      return false;
    }

    String subtype = mediaType.subtype();
    return mediaType.type().equals("application") && (subtype.equals("json") || subtype.startsWith("vnd.") && subtype.endsWith("+json"));
  }
}

File: client-support/src/main/kotlin/io/github/ruedigerk/contractfirst/generator/client/internal/MediaTypes.kt
package io.github.ruedigerk.contractfirst.generator.client.internal

import okhttp3.MediaType
import okhttp3.MediaType.Companion.toMediaTypeOrNull

/**
 * Helper methods for working with media types.
 */
internal object MediaTypes {

  @JvmStatic
  fun isJsonMediaType(mediaType: String?): Boolean {
    return isJsonMediaType(mediaType?.toMediaTypeOrNull())
  }

  @JvmStatic
  fun isJsonMediaType(mediaType: MediaType?): Boolean {
    if (mediaType == null) {
      return false
    }

    val subtype = mediaType.subtype
    return mediaType.type == "application" && (subtype == "json" || subtype.startsWith("vnd.") && subtype.endsWith("+json"))
  }
}
